client/web/src/repo/settings/RepoSettingsOptionsPage.tsx

```diff
@@ -15,90 +13,73 @@ import { SettingsAreaRepositoryFields } from '../../graphql-operations'
 import { eventLogger } from '../../tracking/eventLogger'
 
 import { fetchSettingsAreaRepository } from './backend'
+import { FC, useEffect, useState } from 'react'
 
 interface Props extends RouteComponentProps<{}> {
     repo: SettingsAreaRepositoryFields
 }
 
-interface State {
-    /**
-     * The repository object, refreshed after we make changes that modify it.
-     */
-    repo: SettingsAreaRepositoryFields
-
-    loading: boolean
-    error?: string
-}
-
 /**
  * The repository settings options page.
  */
-export class RepoSettingsOptionsPage extends React.PureComponent<Props, State> {
-    private repoUpdates = new Subject<void>()
-    private subscriptions = new Subscription()
+export const RepoSettingsOptionsPage: FC<Props> = ({ repo: propsRepo }) => {
+    const repoUpdates = new Subject<void>()
+    const subscriptions = new Subscription()
 
-    constructor(props: Props) {
-        super(props)
+    const [loading] = useState<boolean>(false)
+    const [repository, setRepository] = useState<SettingsAreaRepositoryFields>(propsRepo)
+    const [error, setError] = useState<string | undefined>(undefined)
 
-        this.state = {
-            loading: false,
-            repo: props.repo,
-        }
-    }
-
-    public componentDidMount(): void {
+    useEffect(() => {
         eventLogger.logViewEvent('RepoSettings')
-
-        this.subscriptions.add(
-            this.repoUpdates.pipe(switchMap(() => fetchSettingsAreaRepository(this.props.repo.name))).subscribe(
-                repo => this.setState({ repo }),
-                error => this.setState({ error: asError(error).message })
+        subscriptions.add(
+            repoUpdates.pipe(switchMap(() => fetchSettingsAreaRepository(propsRepo.name))).subscribe(
+                repoLocal => setRepository(repoLocal),
+                errorLocal => setError(asError(errorLocal).message)
             )
         )
```

Comment:
I lack specialized knowledge about React, but why do we need `repoUpdates` `Subject` and the `switchMap` here? Since `repoUpdates` is not interacted with outside of this `useEffect` could we instead:

```suggestion
  subscriptions.add(
    fetchSettingsAreaRepository(propsRepo.name).subscribe(
      repoLocal => setRepository(repoLocal),
      errorLocal => setError(asError(errorLocal).message),
    )
  )
```

Reply:
not an expert as well :)
I think Naman mentioned quite the same thing and using a hook here will remove the subscription altogether.

Reply:
👍